### PR TITLE
job details

### LINKED ISSUE #1251: Show stack trace in job selector when job fails
**What would that feature address**
When a job is failing, it is helpful to show the logs in the job selector to understand why the data is incorrect.

***Description of the ideal solution***
Display the job logs in a modal accessible from a job in the list

**Acceptance Criteria**
- [ ] Ensure new code is unit tested, and check code coverage is at least 90%
- [ ] Create related issue in taipy-doc for documentation and Release Notes
- [ ] Check if a new demo could be provided based on this, or if legacy demos could be benefit from it
- [ ] Ensure any change is well documented

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #1251

```
import datetime as dt

import taipy as tp
from taipy import Config, Frequency
from taipy.gui import State


def on_change(state: State, var: str, val):
    print(f"on_change(state: State, var: {var}, val type: {type(val)}, val: {val})")

# Function to run a Dataiku scenario
def run_something(input_1, input_2):
    datetime = dt.datetime.now()
    date = dt.date(2018, 1, 1)
    int_var = 10
    string_var = "String"
    # raise Exception("Fred's exception")
    return datetime, date, int_var, string_var


data = {
    "toto": list(range(10_000)),
    "titi": [2 * i for i in range(10_000)],
    "tata": [4 * i for i in range(10_000)],
}


input_1_cfg = Config.configure_data_node(
    id="input_1_data_node",
    default_data=data,
)


input_2_cfg = Config.configure_data_node(
    id="input_2_data_node",
    default_data=data,
)

datetime_cfg = Config.configure_data_node(id="datetime_data_node")
date_cfg = Config.configure_data_node(id="date_data_node")
int_cfg = Config.configure_data_node(id="int_data_node")
string_cfg = Config.configure_data_node(id="string_data_node")


# Scenario and task configuration in Taipy
scenario_task_cfg = Config.configure_task(
    id="scenario_task",
    function=run_something,
    input=[input_1_cfg, input_2_cfg],
    output=[datetime_cfg, date_cfg, int_cfg, string_cfg],
)

scenario_cfg = Config.configure_scenario(id="scenario", task_configs=[scenario_task_cfg], frequency=Frequency.DAILY)

data_node = None

# GUI Markdown content
scenario_md = """
<|1 1|layout|
<|{scenario}|scenario_selector|>

<|{scenario}|scenario|>
|>

<|{scenario}|>


<|job_selector|>

<|1 1|layout|
Input 1 Dataset:
<|{scenario.input_1_data_node if scenario else None}|data_node|>

Input 2 Dataset:
<|{scenario.input_2_data_node if scenario else None}|data_node|>
|>


<|1 1|layout|
<|{data_node}|data_node_selector|>

<|{data_node}|data_node|>
|>
"""

# Main execution block with GUI setup
if __name__ == "__main__":
    tp.Orchestrator().run()
    scenario = tp.create_scenario(scenario_cfg)
    scenario.add_tag("test")
    scenario1 = tp.create_scenario(scenario_cfg)
    scenario1.add_tag("test")
    # Run the GUI
    tp.Gui(scenario_md).run(title="1251 Job details")

```

`<|job_selector|on_details={False}|>`
`<|job_selector|on_details=on_details_callback|>

on_details_callback(state: State, id: str, payload: {'action': 'on_details_callback', 'args': ['JOB_scenario_task_e2e9329d-2d2c-45fd-b2e0-64921b88a666']})
`

### CODE REVIEW

**Review by dinhlongviolin1 — DISMISSED** [review]

**Review by namnguyen20999 — DISMISSED** [review]

File: taipy/gui_core/viselements.json [hunk 1]
```diff
@@ -539,6 +539,26 @@
                         "type": "str",
                         "default_value": "\"50vh\"",
                         "doc": "The maximum height, in CSS units, of the control."
+                    },
+                    {
+                        "name": "on_details",
+                        "type": "Union[Callback, bool]",
+                        "doc": "The name of a function that is triggered when the details icon is pressed.<br/>The parameters of that function are all optional:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>id (str): the component id.</li>\n<li>payload (<code>dict^</code>): a dict that contains a value for args with the Job Id.</li>\n</ul></br>If False, the icon is not shown.",
```
Comment (FabienLelaquais) [inline taipy/gui_core/viselements.json]:
```suggestion
                        "doc": "The name of a function that is triggered when the details icon is pressed.<br/>The parameters of that function are all optional:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>id (str): the id of the control.</li>\n<li>payload (<code>dict^</code>): a dictionary that contains the Job Id in the value for key <i>args<i>.</li>\n</ul></br>If False, the icon is not shown.",
```

File: taipy/gui_core/viselements.json [hunk 2]
```diff
@@ -207,7 +207,7 @@
                     {
                         "name": "on_submission_change",
                         "type": "Callback",
```
Comment (FabienLelaquais) [inline taipy/gui_core/viselements.json:209]: Why not 'Callable'?
"Callback" is confusing and I can elaborate on that.

File: taipy/gui_core/viselements.json [hunk 3]
```diff
@@ -207,7 +207,7 @@
...
+                        "doc": "The name of the function that is triggered when a submission status is changed.<br/><br/>All the parameters of that function are optional:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>submission (Submission): the submission entity containing submission information.</li>\n<li>details (dict): the details on this Callback's invocation.<br/>\nThis dictionary has the following keys:\n<ul>\n<li>submission_status (str): the new status of the submission (possible values: SUBMITTED, COMPLETED, CANCELED, FAILED, BLOCKED, WAITING, RUNNING).</li>\n<li>job: the Job (if any) that is at the origin of the submission status change.</li>\n<li>submittable_entity: submittable (Submittable): the entity (usually a Scenario) that was submitted.</li>\n</ul>",
```
Comment (FabienLelaquais) [inline taipy/gui_core/viselements.json]:
```suggestion
                        "doc": "The name of the function that is triggered when a submission status is changed.<br/><br/>All the parameters of that function are optional:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>submission (Submission): the submission entity containing submission information.</li>\n<li>details (dict): the details on this callback's invocation.<br/>\nThis dictionary has the following keys:\n<ul>\n<li>submission_status (str): the new status of the submission (possible values are: "SUBMITTED", "COMPLETED", "CANCELED", "FAILED", "BLOCKED", "WAITING", or "RUNNING").</li>\n<li>job: the Job (if any) that is at the origin of the submission status change.</li>\n<li>submittable_entity (Submittable): the entity (usually a Scenario) that was submitted.</li>\n</ul>",
```

File: taipy/gui_core/viselements.json [hunk 4]
```diff
@@ -65,7 +65,7 @@
...
+                        "doc": "The name of the function that is triggered when a scenario is about to be created.<br/><br/>All the parameters of that function are optional:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>id (str): the identifier of the scenario selector.</li>\n<li>payload (dict): the details on this Callback's invocation.<br/>\nThis dictionary has the following keys:\n<ul>\n<li>config: the name of the selected scenario configuration.</li>\n<li>date: the creation date for the new scenario.</li>\n<li>label: the user-specified label.</li>\n<li>properties: a dictionary containing all the user-defined custom properties.</li>\n</ul>\n</li>\n<li>The Callback function can return a scenario, a string containing an error message (a scenario will not be created), or None (then a new scenario is created with the user parameters).</li>\n</ul>",
```
Comment (FabienLelaquais) [inline taipy/gui_core/viselements.json]:
```suggestion
                        "doc": "The name of the function that is triggered when a scenario is about to be created.<br/><br/>All the parameters of that function are optional:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>id (str): the identifier of this scenario selector.</li>\n<li>payload (dict): the details on this callback's invocation.<br/>\nThis dictionary has the following keys:\n<ul>\n<li>config (str): the name of the selected scenario configuration.</li>\n<li>date (datetime): the creation date for the new scenario.</li>\n<li>label (str): the user-specified label.</li>\n<li>properties (dic): a dictionary containing all the user-defined custom properties.</li>\n</ul>\n</li>\n<li>The callback function can return a scenario, a string containing an error message (a scenario will not be created), or None (then a new scenario is created with the user parameters).</li>\n</ul>",
```

**Review by FabienLelaquais — DISMISSED** [review]
👍

File: taipy/gui_core/viselements.json (hunk 2 above)
Comment (FredLL-Avaiga) [reply taipy/gui_core/viselements.json:209]: I will in my next PR

**Review by namnguyen20999 — APPROVED** [review]

**Pull request merged**